go/graph/search.go

```diff
@@ -52,3 +52,62 @@ func (g *Graph) DijkstraSearch(start Node) []Path {
 	}
 	return paths
 }
+
+// Context for an AllPathSearch
+type allPathSearch struct {
+	// Graph
+	Graph *Graph
+	// Current path
+	Path []Edge
+	// Set of node values on the path
+	NodeValues map[*interface{}]bool
+	// Final paths
+	Result []Path
+}
+
+// AllPathSearch returns all the paths from the start node to another
+// node in the graph. The graph can be cyclic but cyclic paths will never be included.
+func (g *Graph) AllPathSearch(start, end Node) []Path {
+	aps := &allPathSearch{
+		Path:       []Edge{},
+		NodeValues: make(map[*interface{}]bool),
```

Comment:
This could be initialized with `, len(g.nodes)` to avoid map reallocation as this explores nodes.